lxd/storage/backend_lxd.go

```diff
@@ -1677,6 +1677,14 @@ func (b *lxdBackend) CreateInstanceFromImage(inst instance.Instance, fingerprint
 		return err
 	}
 
+	// We need to lock this operation to ensure the image is not deleted while instance is being created.
+	unlock, err := locking.Lock(context.TODO(), drivers.OperationLockName("DeleteImage", b.name, drivers.VolumeTypeImage, "", fingerprint))
```

Comment:
You can use `b.state.ShutdownCtx` instead of the `TODO`.

Reply:
Yep, I noticed that, but we use TODO/background everywhere. So my question is whether this should be fixed across entire file? Or am I missing something and this is intentional to prevent operation from stopping if lxd is shutdown?

Reply:
My take on it was always try to use the states context if possible. @tomponline do we miss something here?